Grammar/python.gram

```diff
@@ -203,14 +207,18 @@ with_item[withitem_ty]:
     | e=expression { _PyAST_withitem(e, NULL, p->arena) }
 
 try_stmt[stmt_ty]:
+    | invalid_try_stmt
     | 'try' &&':' b=block f=finally_block { _PyAST_Try(b, NULL, NULL, f, EXTRA) }
     | 'try' &&':' b=block ex[asdl_excepthandler_seq*]=except_block+ el=[else_block] f=[finally_block] { _PyAST_Try(b, ex, el, f, EXTRA) }
 except_block[excepthandler_ty]:
+    | invalid_except_stmt_indent
     | 'except' e=expression t=['as' z=NAME { z }] ':' b=block {
         _PyAST_ExceptHandler(e, (t) ? ((expr_ty) t)->v.Name.id : NULL, b, EXTRA) }
     | 'except' ':' b=block { _PyAST_ExceptHandler(NULL, NULL, b, EXTRA) }
-    | invalid_except_block
-finally_block[asdl_stmt_seq*]: 'finally' ':' a=block { a }
+    | invalid_except_stmt
+finally_block[asdl_stmt_seq*]:
+    | invalid_finally_stmt
+    | 'finally' &&':' a=block { a }
```

Comment:
It looks like this rule has changed from

`'finally' ':' a=block { a }`
`'finally' &&':' a=block { a }`

is that on purpose?

Reply:
Yup, although this is sort of independent: this is for the "missing :" message for finally blocks. I could factor into a different PR but as I am shuffling these already I decided to do it here